### PR TITLE
Compile for WASI

### LINKED ISSUE #49: add wasm32-wasi target
It would be great if we could target WASI. Currently:

```
Camerons-MacBook-Pro:path_abs cameron$ cargo build --target wasm32-wasi
    Updating crates.io index
   Compiling proc-macro2 v1.0.21
   Compiling memchr v2.3.3
   Compiling unicode-xid v0.2.1
   Compiling lazy_static v1.4.0
   Compiling syn v1.0.41
   Compiling serde_derive v1.0.116
   Compiling regex-syntax v0.6.18
   Compiling serde v1.0.116
   Compiling std_prelude v0.2.12
   Compiling thread_local v1.0.1
   Compiling aho-corasick v0.7.13
   Compiling quote v1.0.7
   Compiling regex v1.3.9
   Compiling stfu8 v0.2.4
   Compiling path_abs v0.5.0 (/Users/cameron/rs/path_abs)
error[E0425]: cannot find function `symlink_dir` in this scope
   --> src/dir.rs:308:9
    |
308 |         symlink_dir(&self, &dst).map_err(|err| {
    |         ^^^^^^^^^^^-------------
    |         |
    |         help: try calling `symlink_dir` as a method: `self.symlink_dir(&dst)`

error[E0425]: cannot find function `symlink_file` in this scope
   --> src/file.rs:413:9
    |
413 |         symlink_file(&self, &dst).map_err(|err| {
    |         ^^^^^^^^^^^^-------------
    |         |
    |         help: try calling `symlink_file` as a method: `self.symlink_file(&dst)`

warning: unused import: `OsString`
  --> src/ser.rs:16:23
   |
16 | use std::ffi::{OsStr, OsString};
   |                       ^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error[E0046]: not all trait items implemented, missing: `to_stfu8`
   --> src/ser.rs:150:1
    |
28  |       fn to_stfu8(&self) -> String;
    |       ----------------------------- `to_stfu8` from trait
...
150 | / impl<T> ToStfu8 for T
151 | | where
152 | |     T: Borrow<PathBuf>,
153 | | {
...   |
164 | |     }
165 | | }
    | |_^ missing `to_stfu8` in implementation

error[E0046]: not all trait items implemented, missing: `from_stfu8`
   --> src/ser.rs:167:1
    |
32  |       fn from_stfu8(s: &str) -> Result<Self, stfu8::DecodeError>;
    |       ----------------------------------------------------------- `from_stfu8` from trait
...
167 | / impl<T> FromStfu8 for T
168 | | where
169 | |     T: From<PathBuf>,
170 | | {
...   |
185 | |     }
186 | | }
    | |_^ missing `from_stfu8` in implementation

warning: use of deprecated associated function `std::error::Error::description`: use the Display impl or to_string()
   --> src/lib.rs:338:21
    |
338 |         self.io_err.description()
    |                     ^^^^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default

error: aborting due to 4 previous errors; 2 warnings emitted

Some errors have detailed explanations: E0046, E0425.
For more information about an error, try `rustc --explain E0046`.
error: could not compile `path_abs`

To learn more, run the command again with --verbose.
```

### PR DESCRIPTION
**Pull request by vlakreeh**
Fixes #49.

### CODE REVIEW

File: src/ser.rs [hunk 1]
```diff
@@ -151,6 +153,12 @@ impl<T> ToStfu8 for T
 where
     T: Borrow<PathBuf>,
 {
+    #[cfg(target_os = "wasi")]
```
Comment (vitiral) [inline src/ser.rs:156]: this is identical to unix. I think you can use `cfg(any(...))` for such a case, no?

File: src/ser.rs [hunk 2]
```diff
@@ -168,6 +176,14 @@ impl<T> FromStfu8 for T
 where
     T: From<PathBuf>,
 {
+    #[cfg(target_os = "wasi")]
```
Comment (vitiral) [inline src/ser.rs:179]: ditto `cfg(any(...))`

File: src/ser.rs [hunk 3]
```diff
@@ -252,6 +268,13 @@ mod tests {
     use super::super::{PathDir, PathFile, PathInfo, PathMut, PathOps, PathType};
     use super::*;
 
+    #[cfg(target_os = "wasi")]
```
Comment (vitiral) [inline src/ser.rs:271]: ditto `cfg(any(...))`

**Review by vitiral — CHANGES_REQUESTED** [review]
Looks pretty good, minor nits.

Edit: and thanks for the PR!

File: src/ser.rs (hunk 1 above)
Comment (vlakreeh) [reply src/ser.rs:156]: You're right! I didn't know about `cfg(any())`. I'll quickly change this.

Comment (vlakreeh) [reply src/ser.rs]: You're right! I wasn't aware of `cfg(any())`.

**Review by vitiral — APPROVED** [review]

**Comment by RReverser** [comment]
@vitiral Can you please make a minor release for this?

**Comment by RReverser** [comment]
Actually, wait, this PR is wrong. The file descriptor in the API is supposed to be a preopen directory descriptor, not the descriptor for the target file.

**Pull request merged**